### PR TITLE
Roach egg shit

### PR DESCRIPTION
**Pull request by Wouju**
## About The Pull Request

Makes the roach egg a item/weapon now, allowing it to be implanted, picked up and shoved in a microwave

also makes a new snack, a boiled roach egg

also makes the egg and the snack both tiny items, the latter for consistency, the former so you can implant more eggs in people

## Why It's Good For The Game

There was already code for the roach egg to be implanted into people which i believe is really fucking cool, as such this PR makes it possible by making the roach egg a item/weapon

the roach egg can now also be prepared into a highly nutritious and high in protein snack, for that maintenance food experience

## Changelog
:cl:
add: Added new snack, a boiled roach egg, 5 parts water, 1 roach egg. Shove it all into a microwave/fire barrel at max heat
tweak: roach egg is now a tiny sized item that can be picked up
/:cl:

### CODE REVIEW

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 1]
```diff
@@ -1096,6 +1096,18 @@
 	cooked = TRUE
 	taste_tag = list(INSECTS_FOOD,MEAT_FOOD)
 
+/obj/item/weapon/reagent_containers/food/snacks/roach_egg
+	name = "Boiled Roach Egg"
```
Comment (Firefox13) [inline code/modules/reagents/reagent_containers/food/snacks.dm:1100]:
```suggestion
	name = "boiled roach egg"
```

**Comment by L74814** [comment]
oh, shit. speedmerge now

File: code/game/objects/effects/roaches.dm [hunk 2]
```diff
@@ -1,13 +1,13 @@
 //generic procs copied from obj/effect/alien
-/obj/effect/roach
+/obj/item/weapon/roach
```
Comment (mattroks101) [inline code/game/objects/effects/roaches.dm:2]: Why is it a weapon instead of just being an item?

Comment (Wouju) [reply code/game/objects/effects/roaches.dm:2]: it is one now

~~the true answer is that i kinda forgot if the thing would work without being a weapon~~

**Comment by BitBite112** [comment]
Alright, but how will you be able to make it in maints without salt?

File: code/game/objects/effects/roaches.dm [hunk 3]
```diff
@@ -16,44 +16,45 @@
...
+/obj/item/roach/fire_act(datum/gas_mixture/air, exposed_temperature, exposed_volume)
 	if(exposed_temperature > 300 + T0C)
 		health -= 5
 		healthcheck()
 
-/obj/effect/roach/roach_egg
+/obj/item/roach/roach_egg
 	name = "roach egg"
-	desc = "A cockroach egg. It seems to pulse slightly with an inner life."
+	desc = "A cockroach egg, can be eaten with proper preperation. It seems to pulse slightly with an inner life."
```
Comment (Firefox13) [inline code/game/objects/effects/roaches.dm:37]:
```suggestion
	desc = "A cockroach egg, can be eaten with proper preparation. It seems to pulse slightly with an inner life."
```

**Review by Firefox13 — APPROVED** [review]

**Comment by Wouju** [comment]
> Great pr, but how will you be able to make it in maints without salt?

there isn't any water sources in maint either (aside from water tanks that ocassionaly spawn)

**Comment by BitBite112** [comment]
> > Great pr, but how will you be able to make it in maints without salt?
> 
> there isn't any water sources in maint either (aside from water tanks that ocassionaly spawn)

There is. Water tanks aren't that uncommon and you can also use sinks. I think you can empty fire extinguishers as well, though I'm unsure.

**Comment by Wouju** [comment]
> There is. Water tanks aren't that uncommon and you can also use sinks. I think you can empty fire extinguishers as well, though I'm unsure.

aight fuck it, gonna remove the need for salt and maybe add it back when there's a way to get salt

**Pull request merged**